pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -1019,6 +1020,166 @@ func TestHandlePodResourcesResizeWithSwap(t *testing.T) {
+	admitFunc func(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult
+}
+
+// Admit rejects all pods in the podsToReject list with a matching UID.
+func (a *testPodAdmitHandler) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+	if a.admitFunc == nil {
+		return lifecycle.PodAdmitResult{Admit: true}
+	}
+	return a.admitFunc(attrs)
+}
+
+func TestAllocationManagerAddPod(t *testing.T) {
+	if goruntime.GOOS == "windows" {
+		t.Skip("InPlacePodVerticalScaling is not currently supported for Windows")
+	}
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodVerticalScaling, true)
+
+	const containerName = "c1"
+
+	basePod1 := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{UID: "111", Name: "pod1", Namespace: "ns1"},
+		Spec:       v1.PodSpec{Containers: []v1.Container{{Name: containerName, Image: "i1"}}},
+	}
+	basePod2 := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{UID: "222", Name: "pod2", Namespace: "ns2"},
+		Spec:       v1.PodSpec{Containers: []v1.Container{{Name: containerName, Image: "i2"}}},
+	}
+
+	cpu1Mem1G := v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}
+	cpu2Mem2G := v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("2Gi")}
+
+	newPodWithResources := func(basePod *v1.Pod, resources v1.ResourceList) *v1.Pod {
+		podCopy := basePod.DeepCopy()
+		podCopy.Spec.Containers[0].Resources = v1.ResourceRequirements{Requests: resources}
+		return podCopy
+	}
+
+	pod1v1 := newPodWithResources(basePod1, cpu1Mem1G) // Pod1 with 1 CPU and 1G Memory
+	pod1v2 := newPodWithResources(basePod1, cpu2Mem2G) // Pod1 with 2 CPU and 2G Memory
+	pod2v1 := newPodWithResources(basePod2, cpu1Mem1G) // Pod2 with 1 CPU and 1G Memory
+
+	testCases := []struct {
+		name                            string
+		initialAllocatedResourcesState  map[types.UID]v1.ResourceList
+		activePods                      []*v1.Pod
+		podToAdd                        *v1.Pod
+		admitFunc                       func(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult
+		expectAdmit                     bool
+		expectedAllocatedResourcesState map[types.UID]v1.ResourceList
+	}{
+		{
+			name:                           "pod request matches allocated resources state - no changes in allocated resources state",
+			initialAllocatedResourcesState: map[types.UID]v1.ResourceList{basePod1.UID: cpu1Mem1G, basePod2.UID: cpu1Mem1G},
+			activePods:                     []*v1.Pod{pod1v1, pod2v1},
+			podToAdd:                       pod1v1,
+			admitFunc:                      nil,
+			expectAdmit:                    true,
+			// expected state is the same as initial state
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{basePod1.UID: cpu1Mem1G, basePod2.UID: cpu1Mem1G},
+		},
+		{
+			name:                           "pod request does not match allocated resources state - pod admitted based on allocated resources",
```

Comment:
It took me several read-throughs of this test case's code to understand what we are testing here... is there a way to make it more obvious?

```suggestion
			name:                           "pod resize is infeasible - pod still admitted based on existing allocation",
```